api/blueprints/drbS3.py

```diff
@@ -13,8 +14,44 @@
 
 s3 = Blueprint('s3', __name__, url_prefix='/s3')
 
+def validateToken(func):
+    @wraps(func)
+    def decorator(*args, **kwargs):
+        logger.debug(request.headers)
+
+        headers = {k.lower(): v for k, v in request.headers.items()}
+
+        try:
+            _, loginPair = headers['authorization'].split(' ')
+            loginBytes = loginPair.encode('utf-8')
+            user, password = b64decode(loginBytes).decode('utf-8').split(':')
+        except KeyError:
+            return APIUtils.formatResponseObject(
+                403, 'authResponse', {'message': 'user/password not provided'}
+            )
+
+        dbClient = DBClient(current_app.config['DB_CLIENT'])
+        dbClient.createSession()
+
+        user = dbClient.fetchUser(user)
+
+        if not user or APIUtils.validatePassword(password, user.password, user.salt) is False:
+            return APIUtils.formatResponseObject(
+                401, 'authResponse', {'message': 'invalid user/password'}
```

Comment:
I only learned this myself a few weeks ago, but [as part of the HTTP standard](https://www.rfc-editor.org/rfc/rfc9110.html#status.401), an endpoint returning a 401 response should also add a WWW-Authenticate header describing of what kind of authentication the endpoint accepts (in this case, basic auth). 

I think it will become a little more important since we will potentially be accepting different kinds of auth at different endpoints.

When I needed to implement this, [I extended the FormatResponseObject in this PR to take a header object](https://github.com/NYPL/drb-etl-pipeline/pull/284/files#diff-1927a2a7260bf138319644bceb09095e47b82f13b79db71504801f8705c00394R485), so that is a potential approach you could take in the future.